server.go

```diff
@@ -56,6 +56,7 @@ func (s *Server) CreateStream(id string) *Stream {
 	defer s.mu.Unlock()
 
 	if s.Streams[id] != nil {
+		str.close()
```

Comment:
Thanks for the PR! 

I think this can be avoided altogether, by reordering this function so that we create a stream after checking if it already exists. Something like:
```go
func (s *Server) CreateStream(id string) *Stream {
	s.mu.Lock()
	defer s.mu.Unlock()

	if s.Streams[id] != nil {
		return s.Streams[id]
	}

	str := newStream(s.BufferSize, s.AutoReplay)
	str.run()

	s.Streams[id] = str

	return str
}

```

Reply:
Nice! That's simpler to understand. Changing now.